### DISCUSSION TITLE
An Idea: workflow combining SDXL lcm and turbo gives very good result in 20 steps

### DISCUSSION THREAD

**Discussion by frankyifei** [lllyasviel/Fooocus#1277] [Ideas]
I did some tests recently. In those test, I am using [Concept Sliders](https://github.com/rohitgandikota/sliders) to improve the image, including [fix_hands.pt](https://sliders.baulab.info/weights/xl_sliders/fix_hands.pt), [repair_slider.pt](https://sliders.baulab.info/weights/xl_sliders/repair_slider.pt) and [stylegan_latent1](https://sliders.baulab.info/weights/xl_sliders/stylegan_latent1.pt). And [LCM&TurboMix LoRA](https://civitai.com/models/216190/lcmandturbomix-lora-only-12mb-8-step-sampling-effect-is-superior-to-using-lcm-or-turbo-alone) which gives you both LCM and Turbo, depending on the sampler.

My findings:
1.LCM is good at composition and gives almost good human anatomy compared to SDXL turbo, but the results from LCM always seem too blurry to me. 
2.SDXL turbo lora can finish in 8 steps but it is prone to wrong human anatomy or other obvious errors I can't fix easily. 
3.You don't need negative prompt, sliders lora are enough to help with hands and the image quality. I would rather double the speed by setting cfg at 1.
4. _**Refine the LCM result with Turbo 8 steps give me great result. You can also upscale with SDXL Turbo after LCM, which give me 2x size result in short time. On windows with 3060, roughly 1 minutes.  without upscale only 20 seconds.**_

Recommandation:
I hope this can be integrated in fooocus. I currently using comfyui for this workflow only because of the convenience. 

I combine these two in comfyUI and it gives good result in 20 steps. LCM 12-15 steps and SDXL turbo 8 steps. But for upscale, Fooocus is much better than other solution. Currently you have to manually change settings, but the results are fantastic. 

**This is same seed, Speed Mode 30 steps, LCM 15 steps and Turbo 8 steps, Turbo clearly has some issue with extra limbs.** 
![2023-12-08_20-53-39_7764](https://github.com/lllyasviel/Fooocus/assets/8104251/d273cca6-2cb4-4db8-a58f-8a4d7676380b)
![2023-12-08_20-36-26_4420](https://github.com/lllyasviel/Fooocus/assets/8104251/ad0ed01e-07e0-47b2-9444-c54595cac312)
![2023-12-08_20-35-56_1685](https://github.com/lllyasviel/Fooocus/assets/8104251/9a00b959-f85b-434e-86ce-0991d33a3634)

**Upscale using Turbo from LCM result(8 steps):**
![2023-12-08_20-40-44_3169](https://github.com/lllyasviel/Fooocus/assets/8104251/7736f742-90f2-43f8-81ec-32822bf2f2f2)
**here are some results from LCM to Turbo(LCM 15steps + Turbo 8 steps upscale):**
![2023-12-08_19-02-46_5647](https://github.com/lllyasviel/Fooocus/assets/8104251/901eade3-9f1a-42c7-b973-2e17cb42b57e)

![2023-12-08_19-30-41_2500](https://github.com/lllyasviel/Fooocus/assets/8104251/df8eef72-4acd-4c77-b3f0-567d3b2963e8)

Same seed Turbo to Turbo upscale, it seems over sutured and face is uglier? 
![2023-12-08_19-34-17_6724](https://github.com/lllyasviel/Fooocus/assets/8104251/20517cf5-e210-41de-aafe-91f01489e612)

**Comment by ghost** [comment]
Hi, I know the developer of fooocus, this process looks great, do you have the workflow of comfyui to share?

**Reply by frankyifei** [reply]
Here are two comfyui workflows I recently worked on. Try them and see which one you like. 

![223849_100_00001_](https://github.com/lllyasviel/Fooocus/assets/8104251/b45e5874-a5dd-4f5c-a199-a40ba16509fa)
![223220_100_00001_](https://github.com/lllyasviel/Fooocus/assets/8104251/7365bc78-9a0d-4030-acee-f9f9dc638af4)

**Reply by vumichien** [reply]
Can you share the workflow of comfyui? It looks so good @frankyifei

**Reply by frankyifei** [reply]
there is nothing fancy, just lcm 15 steps for a low quality picture, then upscale to 2x size with sdxl turbo, 8 steps. I can even do this in fooocus but there are some disadvantages because sdxl will mess up some details, especially keen caps. I guess we are all waiting for sdxl tile control net now.  My newer workflow is lcm 15 steps, upscale with sd ultimate upscaler to 2x size using sdxl turbo. then do a low denoise(0.2) pass with sdxl turbo to fix the seams from upscaler. 
 
![192818_257264635054739_00001_](https://github.com/lllyasviel/Fooocus/assets/8104251/d752dc0c-ff04-44e6-8602-a5978ecc574c)

**Reply by vumichien** [reply]
All of the steps you can do by fooocus or you need to use some other tool like Automatic1111 or comfyui. These tools are quite complicated to get start :D

**Reply by frankyifei** [reply]
just download the image and drop it to comfyui
